src/vaults/AsyncVault.sol

```diff
@@ -46,9 +51,9 @@ contract AsyncVault is AsyncRedeemVault, IAsyncVault {
         );
 
         if (tokenId == 0) {
-            SafeTransferLib.safeTransferFrom(asset, owner, asyncManager().escrow(), assets);
+            SafeTransferLib.safeTransferFrom(asset, owner, _poolEscrowProvider.escrow(poolId), assets);
```

Comment:
Ok, I understand we do not break compatibility because `BaseInvestmentManager.escrow()` still exists.